src/components/tools/table-tool/use-model-data-set.ts

```diff
@@ -1,24 +1,21 @@
 import classNames from "classnames";
 import { useCallback, useState } from "react";
-import { useCurrent } from "../../../hooks/use-current";
 import { TableContentModelType } from "../../../models/tools/table/table-content";
 import { ToolTileModelType } from "../../../models/tools/tool-tile";
 
 export const useModelDataSet = (model: ToolTileModelType) => {
-  const modelRef = useCurrent(model);
-  const getContent = useCallback(() => modelRef.current.content as TableContentModelType, [modelRef]);
-  const dataSet = useCurrent(getContent().dataSet);
+  const content = (model.content as TableContentModelType);
```

Comment:
Could you pass in `content` from `TableToolComponent` instead of deriving this from the model again?